src/pieces/assets/assets_command.py

```diff
@@ -38,26 +50,107 @@ class AssetsCommands:
-		
+		with open(CONFIG_FILE, 'r') as f:
+			content = f.read().strip()
+			if content:
+			return json.loads(content)
+			else:
+			print("Config file is empty. Creating a new configuration.")
+			return {}
+		except json.JSONDecodeError:
+		print("Invalid JSON in config file. Creating a new configuration.")
+		return {}
+	    else:
+		print("Config file does not exist. Creating a new configuration.")
+		return {}
+
+	@classmethod
+	def save_config(cls, config):
+            with open(CONFIG_FILE, 'w') as f:
+                json.dump(config, f)
 
-		filepath = export_code_to_file(asset_dict["raw"], asset_dict["name"], asset_dict["language"])
+	@classmethod
+	def config(cls, **kwargs):
+            config = cls.load_config()
+            if 'editor' in kwargs:
+                config['editor'] = kwargs['editor']
+                cls.save_config(config)
+                print(f"Editor set to: {kwargs['editor']}")
+            else:
+                print("Current configuration:")
+                print(f"Editor: {config.get('editor', 'Not set')}")
 
-		print_model_details(asset_dict["name"],asset_dict["created_at"],asset_dict["updated_at"],asset_dict["type"],asset_dict["language"],filepath)
+	@classmethod
+        @check_assets_existence
+        def open_asset(cls, item_index=1, open_in_editor=False, **kwargs):
+            asset_ids = AssetsCommandsApi().assets_snapshot
+            try:
+                cls.current_asset = list(asset_ids.keys())[item_index-1]  # because we begin from 1
+            except IndexError:
+                return show_error("Invalid asset index or asset not found.", "Please choose from the list or use 'pieces list assets'")
+        
+            asset_dict = AssetsCommandsApi.extract_asset_info(AssetsCommandsApi.get_asset_snapshot(cls.current_asset))
+        
+            print_model_details(asset_dict["name"], asset_dict["created_at"], asset_dict["updated_at"], asset_dict["type"], asset_dict["language"])
+        
+            code_content = asset_dict["raw"]
+
+            if not open_in_editor:
+                # Determine the lexer
+                try:
+                    lexer = get_lexer_by_name(asset_dict["language"], stripall=True)
+                except:
+                    lexer = guess_lexer(code_content)
+
+                # Print the code with syntax highlighting
+                formatted_code = highlight(code_content, lexer, TerminalFormatter())
+                print("\nCode content:")
+                print(formatted_code)
+
+            # Check if -e flag is used or open_in_editor is True
+            if kwargs.get('e') or open_in_editor:
+                config = cls.load_config()
+                editor = config.get('editor')
+                if editor:
+                    # Save the code to a temporary file
+                    with tempfile.NamedTemporaryFile(mode='w', delete=False, suffix=f".{asset_dict['language']}") as temp_file:
+                        temp_file.write(code_content)
```

Comment:
It is a good idea to use temp file but we might consider removing it and use the default implementation because the save command

Reply:
made the change to revert back to original/ default implementation in the commit [e53de90](https://github.com/pieces-app/cli-agent/pull/152/commits/e53de90972872fba9cebfafc426011bd4cc8ac86)